src/cart/crt_hg.c

```diff
@@ -327,6 +467,88 @@ crt_hg_addr_lookup(struct crt_hg_context *hg_ctx, const char *name,
 	return rc;
 }
 
+int
+crt_hg_remove_addr(hg_class_t *hg_class, hg_addr_t hg_addr)
+{
+	hg_return_t     ret = HG_SUCCESS;
+
+	D_DEBUG(DB_TRACE, "removing hg addr %" PRIu64 "\n",
+			(uint64_t)hg_addr);
+	D_DEBUG(DB_TRACE, "removing  hg class %" PRIu64 "\n",
+			(uint64_t)hg_class);
+	ret = HG_Addr_set_remove(hg_class, hg_addr);
+	if (ret != HG_SUCCESS) {
+		D_ERROR("HG_Addr_set_remove) failed, hg_ret %d.\n", ret);
+		D_GOTO(out, ret = -DER_HG);
```

Comment:
we should return cart error code from here. this function is called from public api crt_hg_remove_client_id(), and if this function fails we will end up returning HG error code instead of CART error code to the end user